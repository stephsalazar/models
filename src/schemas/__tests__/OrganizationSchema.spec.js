const mongoose = require('mongoose/browser');
const { OrganizationSchema } = require('../')(mongoose);

describe('OrganizationSchema', () => {
  it('should fail validation when fields missing', () => {
    // name && slug (slug: se crea en Talento o API, basÃ¡ndonos en el nombre)
    const doc = new mongoose.Document({}, OrganizationSchema);
    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should pass validation when at least name and slug', () => {
    const doc = new mongoose.Document({
      name: 'organizationName',
      slug: 'name.slug',
    }, OrganizationSchema);
    return doc.validate();
  });
});
