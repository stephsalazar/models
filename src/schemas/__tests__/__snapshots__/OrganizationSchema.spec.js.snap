// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationSchema should fail validation when fields missing 1`] = `
Object {
  "name": [ValidatorError: Path \`name\` is required.],
  "slug": [ValidatorError: Path \`slug\` is required.],
}
`;
